src/west/app/project.py

```diff
@@ -315,6 +317,43 @@ def check_call(self, args, cwd=None):
                 level=log.VERBOSE_VERY)
         subprocess.check_call(args, cwd=cwd)
 
+    def check_output(self, args, cwd=None):
+        cmd_str = util.quote_sh_list(args)
+        log.dbg(f"running '{cmd_str}' in {cwd or os.getcwd()}",
+                level=log.VERBOSE_VERY)
+        return subprocess.check_output(args, cwd=cwd)
+
+    def get_head_branch(self, url: str) -> str:
+        # Get the branch which url's HEAD points to. Requires git 2.8.0
+        # or later. Errors out if it can't, prints a banner if it can.
+        err_msg = (f'failed getting the default branch from {url}; '
+                   'please provide the --manifest-rev option')
+
+        # The output looks like this:
+        #
+        # ref: refs/heads/foo	HEAD
+        # 6145ab537fcb3adc3ee77db5f5f95e661f1e91e6	HEAD
+        #
+        # The '--quiet' option disables printing the URL to stderr.
+        try:
+            output = self.check_output(
+                ('git', 'ls-remote', '--quiet', '--symref', url, 'HEAD')
+            ).decode('utf-8')
+        except subprocess.CalledProcessError:
+            log.die(err_msg)
+
+        for line in output.splitlines():
+            if not line.startswith('ref: '):
+                continue
+            # Per git-check-ref-format(1), references can't have tabs
+            # in them, so this doesn't have any weird edge cases.
+            ret = line[len('ref: '):].split('\t')[0]
+            log.small_banner('no --manifest-rev was given; '
+                             f"using remote's default branch: {ret}")
+            return ret
```

Comment:
Just to be on the safe side:
```suggestion
            if ret.strip():
                   return ret
```

`'ref:    '[len('ref: '):].split('\t')[0]` is pure whitespace and whitespace in an error message is very puzzling.